command/connect/envoy/envoy_test.go

```diff
@@ -510,15 +480,8 @@ func TestGenerateConfig(t *testing.T) {
 		t.Run(tc.Name, func(t *testing.T) {
 			require := require.New(t)
 
-			testDir := testutil.TempDir(t, "envoytest")
-			defer os.RemoveAll(testDir)
-
-			if len(tc.Files) > 0 {
-				for fn, fv := range tc.Files {
-					fullname := filepath.Join(testDir, fn)
-					require.NoError(ioutil.WriteFile(fullname, []byte(fv), 0600))
-				}
-			}
+			tempDir := fs.NewDir(t, t.Name(), fs.WithFiles(tc.Files))
+			defer tempDir.Remove()
```

Comment:
Now that we are on go1.14 we don't need this `defer` because the `t.Cleanup` is automatically registered. I've left it here in case the tests are run on go1.13 or earlier, but in the future we will be able to remove it.